stylecheck/lint.go

```diff
@@ -375,11 +375,20 @@ func CheckErrorStrings(pass *analysis.Pass) (interface{}, error) {
 				if !ok {
 					continue
 				}
-				if !irutil.IsCallToAny(call.Common(), "errors.New", "fmt.Errorf") {
+				if !irutil.IsCallToAny(call.Common(), config.For(pass).ErrorFunctions...) {
 					continue
 				}
 
-				k, ok := call.Common().Args[0].(*ir.Const)
+				var arg ir.Value
+				// If it is a call to a method of a structure,
+				// the first argument is receiver
+				if _, ok := call.Common().Args[0].(*ir.Load); ok {
+					arg = call.Common().Args[1]
```

Comment:
We can't trust users to only provide functions that have the right number of arguments. We need bounds checks, for both the Args[0] and the Args[1] cases.

Reply:
Previously, it is checked that this function has arguments and one of them is `error`.

```
				if !irutil.IsCallToAny(call.Common(), config.For(pass).ErrorFunctions...) {
					continue
				}
```